### PR TITLE
#159053966 Ft favorite articles 

### PR DESCRIPTION
**Pull request by chepmanu**
#### What does this PR do?
Favorite and unfavorite articles.

#### Description of tasks to be completed
An authenticated user should be able to favorite and unfavorite articles.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans.git
```
```sh
$ git checkout -b ft-favorite-articles-159053966
```
```sh
$ cd ah-titans/authors
```

> Add **.env** file and copy **example.env** content.
> Add your local database configuration settings to your **.env** file

```sh
$ source .env
```
```sh
$ python3.6 -m venv venv
```
```sh
$ source venv/bin/activate
```
```sh
$ pip install -r requirements.txt
```
```sh
$ python manage.py makemigrations
```
```sh
$ python manage.py runserver
```

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#159053966](https://www.pivotaltracker.com/story/show/159053966)

#### Screenshots
**One favortied articles UI**
image

**Favorited articles UI**

### CODE REVIEW

**Comment by petermwash** [comment]
@chepmanu Check on how you have written you PR description. What the PR does and the description are not for this particular PR. You should change that.

File: authors/tests/articles/test_favorite_aritcles.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        self.assertEqual(response2.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(True, response.data['favorited'])
+        self.assertEqual(1, response.data['favoriteCount'])
+        self.assertEqual(2, response1.data['favoriteCount'])
+        self.assertEqual(3, response2.data['favoriteCount'])
+
+
+
+
+
```
Comment (petermwash) [inline authors/tests/articles/test_favorite_aritcles.py]: Add an end of line here

File: authors/templates/static/fav-article.html [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        <div class="footer-copyright">
+            <div class="container">
+                © 2018 Created with Love by team titans.
+            </div>
+        </div>
+    </footer>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0-beta/js/materialize.min.js"></script>
+</body>
+
+</html>
```
Comment (petermwash) [inline authors/templates/static/fav-article.html]: Add an end of line.

**Comment by ronaldndirangu** [comment]
Please resolve the current merge conflict.

**Comment by ronaldndirangu** [comment]
Also, add postman screenshots to show the response from the requests you've made.

File: authors/tests/articles/test_favorite_aritcles.py (hunk 1 above)
Comment (janetnim) [reply authors/tests/articles/test_favorite_aritcles.py]: Remove unnecessary lines 323-326

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -1,18 +1,15 @@
-
 from django.db.models import Avg
+from .models import Article
```
Comment (migot01) [inline authors/apps/articles/views.py]: check your order of  imports in views.py

**Review by janetnim — COMMENTED** [review]
in serializers.py
```
favoriteCount = serializers.SerializerMethodField(method_name='get_favorite_count')
```
I think we shouldn't use camel case in naming variables in Python...favorite_count

**Comment by chepmanu** [comment]
@janetnim 
image
The guideline given showed that we had to return it that way.

**Comment by janetnim** [comment]
Oh okay....missed that

**Review by janetnim — APPROVED** [review]

**Review by migot01 — APPROVED** [review]

File: authors/apps/articles/serializers.py [hunk 4]
```diff
@@ -141,4 +160,4 @@ class Meta:
     
     def to_representation(self, obj):
         return obj.tag
-    
+
```
Comment (johnmusiu) [inline authors/apps/articles/serializers.py]: I think there is some trailing white space here, check on that

File: authors/apps/articles/views.py [hunk 5]
```diff
@@ -1,29 +1,15 @@
...
-
-
-
-class LargeResultsSetPagination(PageNumberPagination):
-    """
-    Set pagination results settings
-    """
-    page_size = 10
-    page_size_query_param = 'page_size'
-    max_page_size = 10
```
Comment (johnmusiu) [inline authors/apps/articles/views.py]: @dcDalin I need your eyes on this file, check if your pagination feature was not accidentally dropped

**Review by johnmusiu — CHANGES_REQUESTED** [review]
I need @dcDalin to have a look at `articles/views.py` Pagination feature seems to have been accidentally dropped

**Comment by chepmanu** [comment]
@johnmusiu Thank you, I had not seen that, that might have been dropped when I fixed the conflicts.It is sorted now.

**Review by johnmusiu — APPROVED** [review]

**Pull request merged**